src/bin/pg_autoctl/pgsetup.c

```diff
@@ -641,33 +630,37 @@ bool
 pg_setup_get_local_connection_string(PostgresSetup *pgSetup,
 									 char *connectionString)
 {
-	char *pg_regress_sock_dir = getenv("PG_REGRESS_SOCK_DIR");
+	char pg_regress_sock_dir[MAXPGPATH];
 	char *connStringEnd = connectionString;
+	bool pg_regress_sock_dir_exists = env_exists("PG_REGRESS_SOCK_DIR");
 
 	connStringEnd += sprintf(connStringEnd, "port=%d dbname=%s",
 							 pgSetup->pgport, pgSetup->dbname);
 
+	if (pg_regress_sock_dir_exists &&
+			!get_env_copy("PG_REGRESS_SOCK_DIR", pg_regress_sock_dir, MAXPGPATH)) {
```

Comment:
Style issue: brackets on their own lines, and usually && at the beginning of the line makes it easier to read the code.